test/NuGet.Clients.Tests/NuGet.CommandLine.Test/MSBuildUtilityTest.cs

```diff
@@ -333,6 +333,68 @@ public void GetMsbuildDirectoryFromPath_PATHENVWithQuotes_Succeeds()
             }
         }
 
+        [Theory]
+        [InlineData("arm64", true)]
+        [InlineData("amd64", true)]
+        [InlineData("ARM64", true)]
+        [InlineData("random", false)]
+        public void GetNonArchitectureDirectory_PATHENVWithArchitecture_Succeeds(string architecutre, bool isArchitectureSpecificPath)
+        {
+            if (RuntimeEnvironmentHelper.IsMono)
+            { // Mono does not have SxS installations so it's not relevant to get msbuild from the path.
+                return;
+            }
+
+            using (var vsPath = TestDirectory.Create())
+            {
+                var msBuildNonArchitectureDir = Directory.CreateDirectory(Path.Combine(vsPath, "MSBuild", "Current", "Bin"));
+                var msBuildExeNonArchitecturePath = Path.Combine(msBuildNonArchitectureDir.FullName, "msbuild.exe");
+                File.WriteAllText(msBuildExeNonArchitecturePath, "foo");
+
+                var msBuildArchitectureDir = Directory.CreateDirectory(Path.Combine(msBuildNonArchitectureDir.FullName, architecutre));
+                var msBuildExeArchitecturePath = Path.Combine(msBuildArchitectureDir.FullName, "msbuild.exe");
+                File.WriteAllText(msBuildExeArchitecturePath, "foo");
+
+                // Act;
+                var msBuildPath = MsBuildUtility.GetNonArchitectureDirectory(msBuildExeArchitecturePath);
+
+                // Assert
+                if (isArchitectureSpecificPath)
+                {
+                    Assert.Equal(msBuildNonArchitectureDir.FullName, msBuildPath);
+                }
+                else
+                {
+                    Assert.Equal(msBuildArchitectureDir.FullName, msBuildPath);
+                }
+            }
+        }
+
+        [Fact]
+        public void GetNonArchitectureDirectory_PATHENVWithArchitecture_Throws()
+        {
+            if (RuntimeEnvironmentHelper.IsMono)
```

Comment:
Wouldn't it be better to skip the test? I think this will show up as a successful test run, which is kind of misleading if it's supposed to be not-applicable.

Reply:
Thanks for @rainersigwald's input : the PATH can be set by any of the subscripts that the developer command prompt script calls, owned by different component teams of VS. For this particular case we have [a script that sets the MSBuild path up](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdev.azure.com%2Fdevdiv%2FDevDiv%2F_git%2FVS%3Fpath%3D%2Fsrc%2Fvc%2Fprojbld%2Fvcvars%2Fvsdevcmd%2Fcore%2Fmsbuild.bat.tt%26version%3DGBmain%26line%3D12%26lineEnd%3D12%26lineStartColumn%3D54%26lineEndColumn%3D59%26lineStyle%3Dplain%26_a%3Dcontents&data=05%7C01%7CLiu.Heng%40microsoft.com%7Cd4646232bd0c4e2feeaa08db742e7464%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638231515113561572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=j8S4Y2tCEKvZb%2BIWHLjFt0XjLb0%2B73fNRR%2Bjt4Apyag%3D&reserved=0).
As is today, you can safely cut “arm64” or “amd64” from the end. There may be a time in the future when the MSBuild on PATH is a .NET Core MSBuild and then the relative path would be different.

So my understanding is, we're not 100% sure what would be the changes in the future, so I hope this could catch the runtime issue just in case, when the PATH changed and no msbuild.exe in the non-architecture specific folder. I'm not sure if we will need this or not in the future.

Reply:
I was thinking along the lines of our other tests which, rather than `return`, actually pass a SkipMono attribute to the test.
https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformFactAttribute.cs#L60

Reply:
Sorry that I misunderstood the feedback.. Fixed by skipping mono test in a better way. Thanks!